### PR TITLE
Allowing duplicates in the FileDataSource to be ignored

### PR DESCRIPTION
**Pull request by JeffAshton**
**Requirements**

- [X] I have added test coverage for new or changed functionality
- [X] I have followed the repository's [pull request submission guidelines](../blob/master/CONTRIBUTING.md#submitting-pull-requests)
- [X] I have validated my changes against all supported platform versions

### CODE REVIEW

File: src/LaunchDarkly.ServerSdk/Files/FlagFileDataMerger.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    // transferring its contents into the format used by the feature store.
+    internal sealed class FlagFileDataMerger
+    {
+        private readonly DuplicateKeysHandling _duplicateKeysHandling;
+
+        public FlagFileDataMerger(DuplicateKeysHandling duplicateKeysHandling)
+        {
+            _duplicateKeysHandling = duplicateKeysHandling;
+        }
+
```
Comment (JeffAshton) [inline src/LaunchDarkly.ServerSdk/Files/FlagFileDataMerger.cs]: image

File: src/LaunchDarkly.ServerSdk/Files/FlagFileDataMerger.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            }
+            if (data.Segments != null)
+            {
+                foreach (KeyValuePair<string, JToken> e in data.Segments)
+                {
+                    AddItem(allData, VersionedDataKind.Segments, FlagFactory.SegmentFromJson(e.Value));
+                }
+            }
+        }
+
```
Comment (JeffAshton) [inline src/LaunchDarkly.ServerSdk/Files/FlagFileDataMerger.cs]: image

File: test/LaunchDarkly.ServerSdk.Tests/FlagFileDataTest.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using LaunchDarkly.Client;
+using LaunchDarkly.Client.Files;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace LaunchDarkly.Tests
+{
+    public class FlagFileDataTest
```
Comment (eli-darkly) [inline test/LaunchDarkly.ServerSdk.Tests/FlagFileDataTest.cs]: Trivial: this is really `FlagFileDataMergerTest`

Comment (JeffAshton) [reply test/LaunchDarkly.ServerSdk.Tests/FlagFileDataTest.cs]: Fixed

**Comment by eli-darkly** [comment]
This is included in the 5.11.0 release.

**Pull request merged**